pex/pip.py

```diff
@@ -588,18 +611,179 @@ def create(
                 isolated_pip_builder.info.venv = True
                 for dist_location in third_party.expose(["pip", "setuptools", "wheel"]):
                     isolated_pip_builder.add_dist_location(dist=dist_location)
+                env_var_names = dict(
+                    patched_markers_env_var_name=cls._PATCHED_MARKERS_FILE_ENV_VAR_NAME,
+                    skip_markers_env_var_name=cls._SKIP_MARKERS_ENV_VAR_NAME,
+                    python_versions_env_var_name=cls._PYTHON_VERSIONS_FILE_ENV_VAR_NAME,
+                )
                 with named_temporary_file(prefix="", suffix=".py", mode="w") as fp:
                     fp.write(
                         dedent(
                             """\
                             import os
                             import runpy
+                            import sys
 
+                            skip_markers = os.environ.pop({skip_markers_env_var_name!r}, None)
                             patched_markers_file = os.environ.pop(
                                 {patched_markers_env_var_name!r}, None
                             )
-                            if patched_markers_file:
-                                def patch_markers():
+                            if skip_markers is not None and patched_markers_file is not None:
+                                raise AssertionError(
+                                    "Pex should never both set both {skip_markers_env_var_name} "
+                                    "and {patched_markers_env_var_name} environment variables."
+                                )
+
+                            if skip_markers:
+                                python_full_versions = []
+                                python_versions = []
+                                python_majors = []
+
+                                python_versions_file = os.environ.pop(
+                                    {python_versions_env_var_name!r}, None
+                                )
+                                if python_versions_file:
+                                    import json
+
+                                    with open(python_versions_file) as fp:
+                                        python_full_versions = json.load(fp)
+                                    python_versions = sorted(
+                                        set(
+                                            (version[0], version[1])
+                                            for version in  python_full_versions
+                                        )
+                                    )
+                                    python_majors = sorted(
+                                        set(version[0] for version in  python_full_versions)
+                                    )
+
+                                # 1.) Universal dependency environment marker applicability.
```

Comment:
The 1, 2, 3 starting here is the only magic to get `pip download` working for universal resolves. In short, there are 3 items to deal with:
1. Requires-Dist: Need to only partially evaluate env markers for python_version / python_full_version (if --interpreter-constraint) but not anything else.
2. Wheel tags: Need to ignore these saving for the major / minor version part of the python tag if --interpreter-constraint.
3. Requires-Python: Need to ignore this / substitute info from --interpreter-constraint to select.